common/transform/src/main/java/ai/h2o/mojos/deploy/common/transform/Utils.java

```diff
@@ -23,4 +24,22 @@ public static void copyResultFields(MojoFrame mojoFrame, List<Row> outputRows) {
       }
     }
   }
+
+  /**
+   * Sanitize boolean string literal values true / false (case insensitive) into 1 / 0 respectively.
+   * @return sanitized string.
+   */
+  public static String sanitizeBoolean(String value, DataField.DataTypeEnum dataType) {
+    if (
+        dataType.equals(DataField.DataTypeEnum.FLOAT32)
+            || dataType.equals(DataField.DataTypeEnum.FLOAT64)
+    ) {
+      if (value.equalsIgnoreCase("true")) {
```

Comment:
best to compare against "true" and not `value`, to avoid any potential npe.

Reply:
thx @achraf-mer , see my response:
- our API is built on the swagger schema to parse the request payload. And according to the schema, `Row` is an array of String https://github.com/h2oai/dai-deployment-templates/blob/692de21b35d516e9da8b53d36dc546fcaf16264c/common/swagger/v1/swagger.yaml#L180-L183
Even if we get around with the issue by reversing the order, that is invalid request, even worse that could hide the real problem, if it is going to be a unchecked exception, the best thing to handle is just to fail fast

 - i doubt it would be possible to for our API handler to receive a null value, as all requests coming through monitor proxy which will do the validation work to see the requests are subject to the schema or not, it will exclude all invalid requests, just show you an example in below:

send a `null` value in payload
```
jinghan@JHan-MBP16 h2o % curl -X POST -H "Content-Type: application/json" -d @- https://model.jh-test-1.mlops-dev.h2o.ai/f48f64b1-c3ef-410c-9df7-40098b679563/model/score << EOF
{
  "fields": [
    "timestamp",
    "count",
    "visibility",
    "temperature",
    "windSpeed",
    "humidity",
    "pressure",
    "substation"
  ],
  "rows": [
    [
      null,
      "0",
      "0",
      "0",
      "0",
      "0",
      "0",
      "0"
    ]
  ]
}
EOF
```

exception from monitor proxy
```
2023-04-06 17:32:49,162 [ERROR] akka.actor.ActorSystemImpl - Error during processing of request: 'Expected String as JsString, but got null'. Completing with 500 Internal Server Error response. To change default exception handling behavior, provide a custom ExceptionHandler.                                                                                                        │
│ spray.json.DeserializationException: Expected String as JsString, but got null                                                                                                                                                                                                                                                                                                             │
│     at spray.json.package$.deserializationError(package.scala:23)                                                                                                                                                                                                                                                                                                                          │
│     at spray.json.ProductFormats.fromField(ProductFormats.scala:63)                                                                                                                                                                                                                                                                                                                        │
│     at spray.json.ProductFormats.fromField$(ProductFormats.scala:51)                                                                                                                                                                                                                                                                                                                       │
│     at ai.h2o.monitor.proxy.localrestscorer.LocalRestScorerFormat$.fromField(LocalRestScorerFormat.scala:7)                                                                                                                                                                                                                                                                                │
│     at spray.json.ProductFormatsInstances$$anon$4.read(ProductFormatsInstances.scala:109)                                                                                                                                                                                                                                                                                                  │
│     at spray.json.ProductFormatsInstances$$anon$4.read(ProductFormatsInstances.scala:95)                                                                                                                                                                                                                                                                                                   │
│     at ai.h2o.monitor.proxy.localrestscorer.LocalRestScorerRequest.<init>(LocalRestScorerRequest.scala:11)                                                                                                                                                                                                                                                                                 │
│     at ai.h2o.monitor.proxy.localrestscorer.LocalRestScorer.newScoringRequest(LocalRestScorer.scala:63)                                                                                                                                                                                                                                                                                    │
│     at ai.h2o.monitor.proxy.localrestscorer.LocalRestScorer.newScoringRequest(LocalRestScorer.scala:36)                                                                                                                                                                                                                                                                                    │
│     at ai.h2o.monitor.proxy.ProxyRoute.$anonfun$monitorEndpoint$1(ProxyRoute.scala:34) 
```

Reply:
it does not matter if it can get to to code or not, at least for now, it is generally a good practice to compare with reference to the known string than the variable.

Reply:
@achraf-mer ok, i see your point and i m not insist on not changing that, just fyi it is absolutely violate the request schema if it is a `null` value and with or without monitor proxy, the unchecked exception would be a normal reaction at API layer.
updated 
https://github.com/h2oai/dai-deployment-templates/pull/327/commits/7c9b0ec62ffe42650fa4282eabaed1dfd40f27ce